src/test/java/io/knowledgeassets/myskills/server/user/profile/UserProfileDocumentServiceTests.java

```diff
@@ -34,205 +34,168 @@
-										.build())
-								.currentLevel(1)
-								.desiredLevel(2)
-								.priority(3)
-								.build()
-				));
-
-		final byte[] anonymousUserProfileDocument = userProfileDocumentService.getAnonymousUserProfileDocument("5acc24df-792a-4458-8d01-0c67033eceff");
-		assertThat(anonymousUserProfileDocument).isNotEmpty();
-		try (InputStream is = new ByteArrayInputStream(anonymousUserProfileDocument)) {
-			assertThat(is).isNotNull();
-			XWPFDocument document = new XWPFDocument();
-			assertThat(document).isNotNull();
+		try (InputStream templateInputStream = new ClassPathResource("templates/user-profile.docx").getInputStream()) {
+
+			given(userProfileDocumentTemplateReader.getTemplate()).willReturn(templateInputStream);
+
+			this.userProfileDocumentService = new UserProfileDocumentService(userRepository, userSkillRepository, userProfileDocumentTemplateReader);
+
+			User user = User.builder()
+					.id("adac977c-8e0d-4e00-98a8-da7b44aa5dd6")
+					.referenceId("5acc24df-792a-4458-8d01-0c67033eceff")
+					.userName("johndoe")
+					.firstName("John")
+					.lastName("Doe")
+					.email("john.doe@mail.com")
+					.coach(false)
+					.academicDegree("Diplom-Wirtschaftsinformatiker")
+					.positionProfile("Software Developer")
+					.summary("Developer")
+					.industrySectors(Arrays.asList("Automotive", "Telecommunication"))
+					.specializations(Arrays.asList("IT Consulting", "Software Integration"))
+					.certificates(Collections.singletonList("Java Certified Programmer"))
+					.languages(Collections.singletonList("Deutsch"))
+					.build();
+
+			given(userRepository.findByReferenceId("5acc24df-792a-4458-8d01-0c67033eceff")).willReturn(
+					Optional.of(user)
+			);
+
+			given(userSkillRepository.findByUserIdOrderByCurrentLevelDesc("adac977c-8e0d-4e00-98a8-da7b44aa5dd6"))
+					.willReturn(Arrays.asList(
+							UserSkill.builder()
+									.id(1L)
+									.user(user)
+									.skill(Skill.builder()
+											.id("e441613b-319f-4698-917d-6a4037c8e330")
+											.name("Angular")
+											.description("JavaScript Framework")
+											.build())
+									.currentLevel(2)
+									.desiredLevel(3)
+									.priority(4)
+									.build(),
+							UserSkill.builder()
+									.id(2L)
+									.user(user)
+									.skill(Skill.builder()
+											.id("3d4236c9-d84a-420a-baee-27b263118a28")
+											.name("Spring Boot")
+											.description("Java Framework")
+											.build())
+									.currentLevel(1)
+									.desiredLevel(2)
+									.priority(3)
+									.build()
+					));
+
+			final byte[] anonymousUserProfileDocument = userProfileDocumentService.getAnonymousUserProfileDocument("5acc24df-792a-4458-8d01-0c67033eceff");
+			assertThat(anonymousUserProfileDocument).isNotEmpty();
+			try (InputStream is = new ByteArrayInputStream(anonymousUserProfileDocument)) {
+				assertThat(is).isNotNull();
+				XWPFDocument document = new XWPFDocument();
+				assertThat(document).isNotNull();
```

Comment:
Would it be easily possible - with little effort - to check whether the placeholders have been resolved correctly in the document? For example, by checking if the strings "Angular" and "Spring Boot" at least appear somewhere in the document?

Reply:
Yes. It is possible. I developed [UserProfileDocumentTestUtils](https://github.com/KnowledgeAssets/myskills-server/blob/feature/SKOOP-66/src/test/java/io/knowledgeassets/myskills/server/user/profile/UserProfileDocumentTestUtils.java) for that.

Done.

Reply:
Fantastic! 😃